bin/main.rb

```diff
@@ -8,67 +10,73 @@ def print_board(arr)
   puts "#{arr[6]} | #{arr[7]} | #{arr[8]}"
 end
 
-def evaluate_game
-  true
-end
-
-def update_player(player)
-  player
+def checks_input
+  var = gets.chomp
+  until yield(var)
+    puts 'Please enter a valid option'
+    print 'Input value here: '
+    var = gets.chomp
+  end
+  var
 end
 
 # Game begins
 puts 'Welcome to tic-tac-toe game...'
-puts 'Please choose one of the following options: Select 1 to start a new game, Select 2 to quit'
-option = gets.chomp
 
 # Start New Game
-if option.to_i == 1
-
-  # player input
-  puts 'Please enter the name of player one'
-  player_one = gets.chomp
-
-  puts 'Please enter the name of player two'
-  player_two = gets.chomp
-
-  # start Game
-  puts 'The game has begin'
-  game_not_finished = true
-
-  # Game variables
-  player_current = player_one
-  board = [1, 2, 3, 4, 5, 6, 7, 8, 9]
-
-  while game_not_finished
-
-    # Players turn instruction
+# player input
+puts 'Please enter the name of player one'
+print 'Input value here: '
+name = checks_input { |var| var != '' }
+symbol_arr = ['X', 'O', '!', '?', '$']
+puts "Please enter your choice symbol: #{symbol_arr}"
+print 'Input value here: '
+symbol = checks_input { |var| symbol_arr.include?(var) }
+
+symbol_arr.delete(symbol)
+
+player_one = Player.new(name, symbol)
+
+puts 'Please enter the name of player two'
+print 'Input value here: '
+name = checks_input { |var| var != '' }
+
+puts "Please enter your choice symbol: #{symbol_arr}"
+print 'Input value here: '
+symbol = checks_input { |var| symbol_arr.include?(var) }
+
```

Comment:
You are repeating this code for each player. You should create two separate methods to get the player's name and symbol and then call it here.